src/maui/SSW.Rewards.Mobile/SSW.Rewards.Mobile/ViewModels/QuizViewModel.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.ObjectModel;
+using System.Windows.Input;
+using CommunityToolkit.Mvvm.Messaging;
+using SSW.Rewards.Mobile.Messages;
+
+namespace SSW.Rewards.Mobile.ViewModels
+{
+    public class QuizViewModel : BaseViewModel, IRecipient<QuizzesUpdatedMessage>
+    {
+        private readonly IQuizService _quizService;
+
+        private string quizDetailsPageUrl = "quiz/details";
+
+        public ObservableCollection<QuizDto> Quizzes { get; set; } = new ObservableCollection<QuizDto>();
+
+        public ICommand OpenQuizCommand { get; set; }
+
+        public QuizViewModel(IQuizService quizService)
+        {
+            _quizService = quizService;
+
+            OpenQuizCommand = new Command<int>(
+                execute:
+                async (id) => 
+                {
+                    var quiz = Quizzes.FirstOrDefault(q => q.Id == id);
+                    await OpenQuiz(id, quiz.Icon);
+                },
+                canExecute:
+                (id) =>
+                {
+                    return Quizzes.FirstOrDefault(q => q.Id == id).Passed == false;
+                });
+
+            WeakReferenceMessenger.Default.Register(this);
```

Comment:
At that moment no one sends `QuizzesUpdatedMessage`. Should we still subscribe to it?

Reply:
No we shouldn't subscribe to messages that are not being sent. But we should check to see whether is an error of omission first.